parser/src/lexer/tests/identifiers.rs

```diff
@@ -105,14 +107,25 @@ fn valid_tokenization_indexed_trace_access() {
 fn error_identifier_with_invalid_characters() {
     let source = "enf clk@' = clk + 1";
     // "@" is not in the allowed characters.
-    let expected = Error::ScanError(7..8);
+    let expected = LexicalError::UnexpectedCharacter {
+        start: SourceIndex::UNKNOWN,
```

Comment:
To be able to construct a `SourceIndex`, we need to know the `SourceId` associated with the content that was parsed, and that requires having the `CodeMap` on hand. Additionally, in the vast majority of tests, the actual spans aren't useful/significant, which is why I make AST types that contain spans ignore their span in their `PartialEq` implementation. That said, in this case, I would have preferred to assert the locations like in the previous test, however I wanted to avoid reworking the test helpers too much and increasing the amount of code to review. Instead I verified the existing test behavior by looking at the diagnostics emitted during the test (which show a snippet of the source, with a caret pointing to the location of the error).

But since you raised a question about it, and I already felt a bit weird about leaving it like that, I'll revisit this test and add a helper that also asserts the location so we can preserve all of the original test behavior explicitly.

Reply:
Thanks for the explanation. I think the update with the helper is an improvement, though it was a minor thing anyway